server/sonar-server/src/main/java/org/sonar/server/rule/DefaultRuleFinder.java

```diff
@@ -103,61 +125,97 @@ public DefaultRuleFinder(IndexClient indexes, UserSession userSession) {
+      List<RuleDto> rules = ruleDao.selectByQuery(dbSession, query);
+      if (!rules.isEmpty()) {
+        RuleDto rule = rules.get(0);
+        return toRule(rule, ruleDao.selectRuleParamsByRuleKey(dbSession, rule.getKey()));
+      }
       return null;
+    } finally {
+      dbClient.closeSession(dbSession);
     }
   }
 
   @Override
   public final Collection<org.sonar.api.rules.Rule> findAll(org.sonar.api.rules.RuleQuery query) {
-    List<org.sonar.api.rules.Rule> rules = newArrayList();
-    for (Rule rule : index.search(toQuery(query), new QueryContext(userSession)).getHits()) {
-      rules.add(toRule(rule));
+    DbSession dbSession = dbClient.openSession(false);
+    try {
+      List<RuleDto> rules = ruleDao.selectByQuery(dbSession, query);
+      if (rules.isEmpty()) {
+        return Collections.emptyList();
+      }
+      return convertToRuleApi(dbSession, rules);
+    } finally {
+      dbClient.closeSession(dbSession);
+    }
+  }
+
+  private Collection<org.sonar.api.rules.Rule> convertToRuleApi(DbSession dbSession, List<RuleDto> ruleDtos) {
+    List<org.sonar.api.rules.Rule> rules = new ArrayList<>();
+    List<RuleKey> ruleKeys = FluentIterable.from(ruleDtos).transform(RuleDtoToKey.INSTANCE).toList();
+    List<RuleParamDto> ruleParamDtos = ruleDao.selectRuleParamsByRuleKeys(dbSession, ruleKeys);
+    ImmutableListMultimap<Integer, RuleParamDto> ruleParamByRuleId = FluentIterable.from(ruleParamDtos).index(RuleParamDtoToRuleId.INSTANCE);
+    for (RuleDto rule : ruleDtos) {
+      rules.add(toRule(rule, ruleParamByRuleId.get(rule.getId())));
     }
     return rules;
   }
 
-  private org.sonar.api.rules.Rule toRule(Rule rule) {
+  private org.sonar.api.rules.Rule toRule(RuleDto rule, List<RuleParamDto> params) {
+    String severity = rule.getSeverityString();
+    String description = rule.getDescription();
+    RuleDto.Format descriptionFormat = rule.getDescriptionFormat();
+
     org.sonar.api.rules.Rule apiRule = new org.sonar.api.rules.Rule();
     apiRule
-      .setName(rule.name())
-      .setLanguage(rule.language())
-      .setKey(rule.key().rule())
-      .setConfigKey(rule.internalKey())
+      .setName(rule.getName())
+      .setLanguage(rule.getLanguage())
+      .setKey(rule.getRuleKey())
+      .setConfigKey(rule.getConfigKey())
       .setIsTemplate(rule.isTemplate())
-      .setCreatedAt(rule.createdAt())
-      .setUpdatedAt(rule.updatedAt())
-      .setDescription(rule.htmlDescription())
-      .setRepositoryKey(rule.key().repository())
-      .setSeverity(rule.severity() != null ? RulePriority.valueOf(rule.severity()) : null)
-      .setStatus(rule.status().name())
-      .setTags(rule.tags().toArray(new String[rule.tags().size()]))
-      .setId(((RuleDoc) rule).id());
+      .setCreatedAt(new Date(rule.getCreatedAt()))
+      .setUpdatedAt(new Date(rule.getUpdatedAt()))
+      .setRepositoryKey(rule.getRepositoryKey())
+      .setSeverity(severity != null ? RulePriority.valueOf(severity) : null)
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Remove this use of "RulePriority"; it is deprecated. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)